app/controllers/sample_transfers_controller.rb

```diff
@@ -12,4 +12,49 @@ def index
     @sample_transfers = perform_pagination(@sample_transfers)
     @sample_transfers = @sample_transfers.map { |t| SampleTransferPresenter.new(t, @navigation_context) }
   end
+
+  def create
+    new_owner = Institution.find_by(uuid: params["institution_id"])
+    if new_owner.nil?
+      flash[:error] = "Destination Institution does not exists"
+      render json: { status: :error }, status: 404
+    else
+      if create_transfer(new_owner, params["samples"])
+        flash[:success] = "All samples have been transferred successfully."
+        render json: { status: :ok }
+      else
+        flash[:error] = "Samples transfer failed."
+        render json: { status: :error }, status: 500
+      end
+    end
+  end
+
+  def confirm
+    transfer = SampleTransfer.find(params[:sample_transfer_id])
+
+    transfer.confirm_and_apply!
```

Comment:
There is no validation that the current user has any access to the transferred sample, without even talking about the permission to do this action.

Maybe duplicate the `authorize_resource?(to_transfer, UPDATE_SAMPLE)` from the `#create_transfer` method below?

Reply:
I added this in [f559b3a](https://github.com/instedd/cdx/pull/1495/commits/f559b3a9bb377b3419146ee1c2502a1108e655e8)

Reply:
Oh, that's great! If we could just extract an `#authorize_confirm?(transfer)` method that would be perfect for now.

Reply:
Yes, I realized that should better be extracted because we need it again for showing the confirm button.